legacy/firmware/layout2.c

```diff
@@ -581,11 +581,18 @@ static bool formatComputedFeeRate(uint64_t fee, uint64_t tx_weight,
   // Compute fee rate and modify it in place for the bn_format_amount()
   // function. We multiply by 100, because we want bn_format_amount() to display
   // two decimal digits.
-  uint64_t fee_rate_multiplied = div_round(100 * fee, tx_size);
-
-  size_t length =
-      bn_format_amount(fee_rate_multiplied, parentheses ? "(" : NULL,
-                       segwit ? " sat/vB" : " sat/B", 2, output, output_length);
+  int decimals = 2;
+  int multiplier = 100;
+  if (multiplier * fee < fee) {
```

Comment:
Not sure if it covers all overflow cases, e.g.:
```python
>>> x = 1<<61; y = (100*x)%(1<<64); (x/1e15, y/1e15, y < x)
(2305.843009213694, 9223.372036854777, False)
```

Maybe it's simpler to change the `if` to use 0 decimals if the fee is above some large constant (e.g. `MAX_UINT64 / 100`)?